src/wizards/validators.ts

```diff
@@ -0,0 +1,76 @@
+// ------------------ Cluster validators
+
+export async function validateBroker(broker: string): Promise<string | undefined> {
+    return validateFieldRequired('Broker', broker);
+}
+
+export async function validateClusterName(cluster: string, existingClusterNames: string[]): Promise<string | undefined> {
+    const result = validateFieldRequired('Cluster name', cluster);
+    if (result) {
+        return result;
+    }
+    return validateFieldUniqueValue('Cluster name', cluster, existingClusterNames);
+}
+
+export async function validateAuthentificationUserName(userName: string): Promise<string | undefined> {
+    return validateFieldRequired('User name', userName);
+}
+
+// ------------------ Topic validators
+
+const maxNameLength = 249
+const legalChars = /^[a-zA-Z0-9\\.\\-]*$/;
+
+export async function validateTopicName(topic: string, existingTopicNames: string[]): Promise<string | undefined> {
+    // See topic name validation rule at https://github.com/apache/kafka/blob/8007211cc982d8458223e866c1ee7d94b69e0249/core/src/main/scala/kafka/common/Topic.scala#L33
+    const result = validateFieldRequired('Topic', topic);
+    if (result) {
+        return result;
+    }
+    else if (topic == "." || topic == "..") {
+        return "Topic cannot be \".\" or \"..\"";
+    }
+    else if (topic.length > maxNameLength) {
+        return `Topic is illegal, can't be longer than ${maxNameLength} characters`;
+    }
+    else if (!legalChars.test(topic)) {
+        return `Topic ${topic} is illegal, contains a character other than ASCII alphanumerics, '.', '_' and '-'`;
+    }
+    return validateFieldUniqueValue('Topic', topic, existingTopicNames);
+}
+
+export async function validatePartitions(partitions: string): Promise<string | undefined> {
+    const result = validateFieldRequired('Partitions', partitions);
+    if (result) {
+        return result;
+    }
+    return validateFieldPositiveNumber('Partitions', partitions);
+}
+
+export async function validateReplicationFactor(replicationFactor: string): Promise<string | undefined> {
+    return validateFieldPositiveNumber('Replication Factor', replicationFactor);

```

Comment:
apparently you can't use a replication factor higher than the cluster's `offsets.topic.replication.factor` value.

IMHO if `offsets.topic.replication.factor`=1, then there's no point in even showing that part of the wizard.

See https://github.com/jlandersen/vscode-kafka/blob/824a615fd5d49e2021b485d694dc9706429bc1a6/src/commands/topics.ts#L145-L156 as an example to get the cluster's config value:

Reply:
> apparently you can't use a replication factor higher than the cluster's offsets.topic.replication.factor value.

have you a link about that?

Reply:
See https://github.com/jlandersen/vscode-kafka/issues/64

Reply:
I dont have a link. This is what I inferred from experience.